attributes/default.rb

```diff
@@ -38,8 +38,8 @@
 default['hopsworks']['db']                       = "hopsworks"
 
 # Usernames and passwords of non-superusers in MySQL
-default['hopsworks']['mysql']['user']['kafka']               = "kafka"
-default['hopsworks']['mysql']['password']['kafka']            = "kafka"
+default['kafka']['mysql']['user']                = "kafka"
```

Comment:
Also I would define the Kafka attributes in the Kafka cookbook and then use them in Hopsworks. It's the natural place to look for them IMHO.

Reply:
Actually now that I think about it. How is it possible that you don't get a circular dependency issue? (`hopsworks` imports `kafka` attributes which import `hopsworks` attributes)

Reply:
Fixed the group name.

Yes I will move them to kafka-cookbook, for sure it's better. 

I don't know why it doesn't complain. Do you know why we need all these here? https://github.com/logicalclocks/hopsworks-chef/blob/1.2/metadata.rb#L27

Reply:
The reason we have it in hopsworks-chef actually is because I did not want to have hopsworks::default depend on kafka-cookbook

Reply:
> Fixed the group name.
> 
> Yes I will move them to kafka-cookbook, for sure it's better.
> 
> I don't know why it doesn't complain. Do you know why we need all these here? https://github.com/logicalclocks/hopsworks-chef/blob/1.2/metadata.rb#L27

They are for Berks which uses them to fetch the dependencies.You could potentially have stuff specified in `metadata.rb` but not in the Berksfile. In that case, Berks will fetch it from the Supermarket.

Reply:
> The reason we have it in hopsworks-chef actually is because I did not want to have hopsworks::default depend on kafka-cookbook

They are 2 different types of dependencies:

* `hopsworks::default` - `kafka::default` is a Karamel dependencies. Which specifies the order of the execution.

* the `depends` in `metadata.rb` specifies the code dependency between cookbook. In our case  `hopsworks-chef` can use code/attributes specified in `kafka-chef` but the other way around should not happen.